tests/functional/test_chain.py

```diff
@@ -159,3 +159,12 @@ def test_set_pending_timestamp_failure(chain):
 def test_set_balance(chain, owner):
     with pytest.raises(APINotImplementedError):
         chain.set_balance(owner, "1000 ETH")
+
+
+def test_get_contract_by_abi(chain, vyper_contract_instance):
```

Comment:
so we have basically this exact test right here: https://github.com/ApeWorX/ape/blob/main/tests/functional/test_contract.py#L13

and also here:
https://github.com/ApeWorX/ape/blob/main/tests/functional/test_contracts_cache.py#L91

i think we need to still re-create, even if mocking required, the new condition you are adding

Also,